site/public/ts/shared/controllers/RemoteController.ts

```diff
@@ -96,5 +97,16 @@ export const RemoteController = (() => {
                 };
             });
         },
+        DeleteUserCircuit(metadata: CircuitMetadata, callback: (result: boolean) => Promise<void> | void = Promise.resolve): void {
+            Chain(async (data: RemoteData) => {
+                const result = await DeleteUserCircuit(data.authState, metadata.getId());
+                callback(result);
+                if (data.metadata !== undefined && data.metadata.getId() === metadata.getId()) {
+                    return {
+                        metadata: metadata.buildOn().withId("").build()
+                    }
+                }
+            });
```

Comment:
I believe deletion should require an auth state, so set the `requireAuth` parameter true when calling chain and it'll take care of the behavior of someone trying to delete without an auth state (which probably isn't something that's going to happen but better safe than sorry)